File: src/year2015/day01/part1.cc
#include <cstdio>
#include <cstdlib>
#include <stdexcept>
#include <string>

#include "../solutions.h"

auto Year2015::Day01::Part1(const std::string& input) -> std::string {
  int floor = 0;
  for (const auto& chr : input) {
    switch (chr) {
      case '(':
        floor++;
        break;
      case ')':
        floor--;
        break;
      default:
        throw std::runtime_error(
            "Invalid character encountered while reading input.txt");
    }
  }
  return std::to_string(floor);
}

File: src/year2015/day01/part1.rs
pub fn part1(input: String) -> Result<String, String> {
  let mut floor = 0;
  for chr in input.bytes() {
    match chr {
      b'(' => floor += 1,
      b')' => floor -= 1,
      _ => {
        return Err("Invalid character encountered while reading input.txt".to_string());
      }
    }
  }
  return Ok(floor.to_string());
}
